### PR TITLE
Added option to creat https url, and to encode it

### PR DESCRIPTION
**Pull request by mipli**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -200,13 +200,19 @@ ASS.prototype.getUrl = function (endpoint, http) {
...
-	var url = this.getUrl(printf('/users/%s/images/%s.jpg', this.username, id), true);
+ASS.prototype.createImageUrl = function (id, actions, options) {
+	options = options ? options : {};
+	options.https = typeof options.https === 'boolean' ? options.https : false;
+	options.encode = typeof options.encode === 'boolean' ? options.encode : false;
+	var url = this.getUrl(printf('/users/%s/images/%s.jpg', this.username, id), !options.https);
 
 	if (actions) {
-		url += '?' + qs.stringify({ t: actions }, { encode: false });
+		url += '?' + qs.stringify({ t: actions }, { encode: options.encode });
```
Comment (pahen) [inline index.js]: I think you must create the signature on the decoded URL if I remember correct.

Comment (pahen) [reply index.js]: See https://github.com/aptoma/dredition-frontend/blob/develop/src/app/services/ass/ass.service.js#L269

Comment (mipli) [reply index.js]: Fixed now

Comment (martinj) [reply index.js]: Signatures are validated on decoded params https://github.com/aptoma/ass/blob/develop/routes/middleware/auth.js#L94

Comment (mipli) [reply index.js]: Yeah, realized that so rewrote it to create the signature based on the un-encoded url. Worked fine now when I tested it locally.

File: index.js [hunk 2]
```diff
@@ -200,16 +200,21 @@ ASS.prototype.getUrl = function (endpoint, http) {
...
-	var url = this.getUrl(printf('/users/%s/images/%s.jpg', this.username, id), true);
+ASS.prototype.createImageUrl = function (id, actions, options) {
+	options = options ? options : {};
+	options.https = typeof options.https === 'boolean' ? options.https : false;
+	options.encode = typeof options.encode === 'boolean' ? options.encode : false;
 
-	if (actions) {
-		url += '?' + qs.stringify({ t: actions }, { encode: false });
-	}
+	var baseUrl = this.getUrl(printf('/users/%s/images/%s.jpg', this.username, id), !options.https);
```
Comment (pahen) [inline index.js]: Wouldn't it be easier to use `encodeURI` as we do in the frontend instead of running `qs.stringify` multiple times?

Comment (mipli) [reply index.js]: For some reason I thought that we didn't have access to `encodeURI` in node. Changing it now.

Comment (mipli) [reply index.js]: Changed now

**Pull request merged**